osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/health/DataSourceHealthCheck.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.shared.health;
+
+import java.sql.SQLException;
+import java.util.Optional;
+import javax.sql.DataSource;
+import org.opensmartgridplatform.shared.config.AppHealthEnabledCondition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Component;
+
+@Component
+@Conditional(AppHealthEnabledCondition.class)
+public class DataSourceHealthCheck implements HealthCheck {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataSourceHealthCheck.class);
+
+  private final DataSource dataSource;
+
+  public DataSourceHealthCheck(final Optional<DataSource> optionalDataSource) {
+    if (optionalDataSource.isPresent()) {
+      this.dataSource = optionalDataSource.get();
+    } else {
+      this.dataSource = null;
+      LOGGER.info("No datasource found for HealtCheck");
+    }
+  }
+
+  @Override
+  public HealthResponse isHealthy() {
+    try {
+      if (this.dataSource != null) {
+        this.dataSource.getConnection();
```

Comment:
It might be good to run the isValid check on the connection to make sure it is not closed.